lib/jsonapi_parameters/stack_limit.rb

```diff
@@ -0,0 +1,44 @@
+module JsonApi
+  module Parameters
+    LIMIT = 3
+
+    class StackLevelTooDeep < StandardError
+    end
+
+    def stack_limit=(val)
+      @stack_limit = val
+    end
+
+    def stack_limit
+      @stack_limit ||= LIMIT
+    end
+
+    def reset_stack_limit!
+      @stack_limit = LIMIT
+    end
+
+    private
+
+    def increment_stack_level!
+      @current_stack_level ||= 0
```

Comment:
what do you think about setting the initial value of `current_stack_level` in the initializer?

Reply:
Related to this: https://github.com/visualitypl/jsonapi_parameters/pull/43#discussion_r468588111